### PR TITLE
Change the updateWidget function to ingest from a byte array

### PR DESCRIPTION
**Pull request by jeffreyli0312**
Declared example transmit frame on on_updateWidget_clicked() below:
7E | 00 19 | 10 | 01 | 00 13 A2 00 41 B1 6D 1C | FF FE | 00 | 00 | 53 45 4E 54 20 46 52 4F 4D 20 42 | D1

decoded QByteArray into:
- start deliminator
- frame length, message length
- frame type, frame ID
- 64 bit address
- message
- and check sum

### CODE REVIEW

File: src/mainwindow/mainwindow.cpp [hunk 1]
```diff
@@ -63,31 +67,170 @@ MainWindow::~MainWindow()
...
+    quint8 frame_length = encoded_msg[2]; // READ THE LENGTH OF FRAME
+    qDebug() << "Frame length: ";
+    qDebug() << frame_length; // outputs 25
+    uint length_of_message = frame_length - 14; // STORE THE ACTUAL LENGTH OF MESSAGE INTO VARIABLE
+    qDebug() << "Length of message: ";
+    qDebug() << length_of_message;
+
+    quint8 frame_type = encoded_msg[3]; // 90 or 8B
+    qDebug() << "Frame type: ";
+    qDebug() << frame_type; // 90 -> 144, 8B -> 139 (to binary)
```
Comment (jeffreyli0312) [inline src/mainwindow/mainwindow.cpp]: Leaving qDebug() output function for testing output, will remove after testing decoding works

**Comment by jeffreyli0312** [comment]
Mavlink_groundside_decoder has no references in the entire project, I removed it from updateWidget. Is it okay to delete the Mavlink_groundside_decoder function?

**Review by rayjinghaolei — APPROVED** [review]

**Comment by rayjinghaolei** [comment]
> Mavlink_groundside_decoder has no references in the entire project, I removed it from updateWidget. Is it okay to delete the Mavlink_groundside_decoder function?

I think it is fine

**Comment by rayjinghaolei** [comment]
@Shrinjay  I don't seem to have the write access to this repo, I guess you will need to do the final confirm and merge

File: src/mainwindow/mainwindow.cpp [hunk 2]
```diff
@@ -63,31 +67,170 @@ MainWindow::~MainWindow()
...
+}
+
 /**
  * @brief Slot to update the GUI with data received from the plane; for CV data, it also outputs to the POGI file
  *
  * @param encoded_msg The complete message as parsed by the handleSerialRead signal
  *
  */
-void MainWindow::updateWidget(QByteArray encoded_msg)
+void MainWindow::updateWidget(QByteArray encoded_msg) // ex: 7E | 00 19 | 10 | 01 | 00 13 A2 00 41 B1 6D 1C | FF FE | 00 | 00 | 53 45 4E 54 20 46 52 4F 4D 20 42 | D1
```
Comment (Shrinjay) [inline src/mainwindow/mainwindow.cpp]: Move this comment into the next line

File: src/mainwindow/mainwindow.cpp [hunk 3]
```diff
@@ -270,6 +413,12 @@ void MainWindow::on_setWaypointNumberButton_clicked()
 
 void MainWindow::on_sendInfoButton_clicked()
 {
+
```
Comment (Shrinjay) [inline src/mainwindow/mainwindow.cpp]: Don't add extra whitespace like this, makes code harder to read

File: src/mainwindow/mainwindow.ui [hunk 4]
```diff
@@ -432,30 +432,37 @@
...
        <widget class="QPushButton" name="pushButton_2">
         <property name="text">
          <string>Set POGI JSON File</string>
         </property>
        </widget>
       </item>
-      <item row="3" column="0" colspan="2">
-       <widget class="QLabel" name="label_19">
+      <item row="6" column="0">
+       <widget class="QPushButton" name="updateWidget">
```
Comment (Shrinjay) [inline src/mainwindow/mainwindow.ui]: Following with the above comment, update the name of this button to testUpdateWidget

File: src/mainwindow/mainwindow.cpp [hunk 5]
```diff
@@ -63,31 +67,170 @@ MainWindow::~MainWindow()
 // SLOTS
 ///////////////////////////////////////////////////////////
 
+void MainWindow::on_updateWidget_clicked()
```
Comment (Shrinjay) [inline src/mainwindow/mainwindow.cpp]: Let's avoid confusion and make it clear this is only for testing, change this function to onTestUpdateWidget_clicked and update the button in the .ui file accordingly.

**Review by Shrinjay — CHANGES_REQUESTED** [review]
Overall, really, really good code, I'm very impressed. Some minor details I noticed but otherwise we're good to merge.

File: src/mainwindow/mainwindow.cpp [hunk 6]
```diff
@@ -15,20 +15,64 @@
...
+#include <QCamera>
+#include <QCameraInfo>
+
+#include <QScopedPointer>
+#include <QVBoxLayout>
+#include <QAction>
+#include <QCameraImageCapture>
+#include <QCameraViewfinder>
+
+void MainWindow::on_testVideo_clicked()
```
Comment (Shrinjay) [inline src/mainwindow/mainwindow.cpp]: Can we quickly take the video stuff out of this PR?

**Review by Shrinjay — CHANGES_REQUESTED** [review]
Let's just quickly get rid of the video stuff

**Pull request merged**